src/Animated.test.js

```diff
@@ -0,0 +1,106 @@
+import Animated, { Easing } from './Animated';
+import ReanimatedModule from './ReanimatedModule';
+import AnimatedNode from './core/AnimatedNode';
+
+jest.mock('./ReanimatedEventEmitter');
+jest.mock('./ReanimatedModule');
+jest.mock('./derived/evaluateOnce');
+
+const { Value, timing, spring, decay } = Animated;
+describe('Reanimated backward compatible API', () => {
+  beforeEach(() => {
+    let nodesCreated = 0;
+    ReanimatedModule.createNode = () => nodesCreated++;
+    ReanimatedModule.dropNode = () => nodesCreated--;
+    ReanimatedModule.getNumberOfNodes = () => nodesCreated;
+  });
+
+  const checkIfAttachAndDetachNodesProperly = jest.fn(animation => {
+    const transX = new Value(0);
+
+    const initial = ReanimatedModule.getNumberOfNodes();
+    const v = new AnimatedNode({ type: 'sampleView', value: 0 });
+    transX.__addChild(v);
+    const before = ReanimatedModule.getNumberOfNodes();
+    const anim = animation.node(transX, animation.config);
+    anim.start();
+    const during = ReanimatedModule.getNumberOfNodes();
+    anim.__value_testOnly.__setAnimation(null, true);
```

Comment:
IMO this test does not verify much. You could just call enough methods here to make the test pass. How can we be sure these methods are going to be called in real life scenario (especially `__value_testOnly.setAnimation(null, true)`)

We need a test that verifies:
 1) that nodes get detached correctly if we have mounted component that uses value that we start an animation for using imperative API (we don't need to test all possible animation types for that)
 2) nodes get detached correctly if that component get unmounted while the animation is running and that finished callback will get called
 3) nodes get detached correctly if we have two components with a single value we start an animation for, then only one these components get unmounted (the value should stay attached)